list-sensors.cpp

```diff
@@ -15,106 +15,216 @@
 // Bus handler singleton
 static sdbusplus::bus::bus systemBus = sdbusplus::bus::new_default();
 
-const std::string prefix = "xyz.openbmc_project.Sensor.Value.Unit.";
-const int prefix_len = prefix.length();
+using PropertyValue = std::variant<int64_t, std::string, bool>;
+using PropertyName = std::string;
+using PropertiesMap = std::map<PropertyName, PropertyValue>;
 
 /**
- * @brief Returns unit shortname by DBus unit name
- *
- * @param dbus_unit - DBus unut name
- * @return unit shortname
+ * @brief Gives a simple access to sensor properties.
  */
-std::string get_unit_shortname(const std::string& dbus_unit)
+class Properties : public PropertiesMap
 {
-    if (dbus_unit.compare(0, prefix_len, prefix) != 0)
+  public:
+    using PropertiesMap::PropertiesMap;
+
+    /**
+     * @brief Check is specified alarm property has true value
+     *
+     * @param name - Proprety name
+     */
+    bool isAlarmed(const PropertyName& name) const
     {
-        return "Unknown";
+        auto it = this->find(name);
+        return (it == this->end() ? false : std::get<bool>(it->second));
```

Comment:
If there is no such property, it means that this alarm type is impossible for this sensor.
Otherwise, we return the actual state.
What makes you disliked?

Reply:
I don't like this whole method, not just this line or what it returns.
This is a class method, but the way it is implemented now requires the caller to have knowledge of the names of D-Bus properties. That doesn't look like a true incapsulated class to me. This method looks like a macro, not like a true C++ class method. That's what I dislike.

Reply:
I renamed both.